### ISSUE TITLE
Post Loggly messages as background file transfers

### ISSUE THREAD

**Issue by rnapier** [melke/SlimLogger#10]
The current implementation of `SwiftLogglyDestination` will lose log messages if there is no network connection when it wants to push the logs, or if the program is terminated or crashes before it posts the logs. The crash case is probably the time you most want the logs.

I'm reworking the destination to write to log files, which are then handed off to `NSURLSessionUploadTask` to upload these files in the background when the system can and feels it's appropriate. See https://github.com/pushd/BackgroundUpload-CocoaLumberjack for an example of this approach for Lumberjack.

Would you prefer this as a separate repository, a new destination (`SwiftLogglyBackgroundDestination`? getting a bit wordy, but consistent), or as a replacement for `SwiftLogglyDestination`? I'm adapting this from the current `SwiftLogglyDestination` code, and plan to use the same fields, etc (so there would be a lot of code duplication that may need to be refactored out).

**Comment by melke** [comment]
Hi @rnapier Sorry for answering so late. Thanks for finding this problem! Updating the existing SwiftLogglyDestination seems like best idea. If you don't mind, please create a PR with this change and I will merge it!

**Comment by rnapier** [comment]
In the end, I wound up rewriting most of SlimLogger to make it more closely fit what I wanted in a logger (structured logs, where you pass a static "message" and a dynamic dictionary of "data"). So my new LogglyDestination won't drop in properly because I changed the `LogDestination` protocol. That said, since I based it heavily on your code, I'm happy to give it back in its current form:

https://gist.github.com/rnapier/7fd71d5c8c767cecac747b87680b8cfd

Thanks for your library. It was a great starting point. Ever since I used [Sirupsen/logrus](https://github.com/Sirupsen/logrus), I've just been a huge fan of structured logging.
